src/fu-tool.c

```diff
@@ -131,6 +131,26 @@ fu_util_stop_daemon (GError **error)
 	return val != NULL;
 }
 
+static gboolean
+fu_util_save_current_state (FuUtilPrivate *priv, GError **error)
+{
+	g_autoptr(GPtrArray) devs = NULL;
+	g_autofree gchar *state = NULL;
+	g_autofree gchar *dirname = NULL;
+	g_autofree gchar *filename = NULL;
+
+	devs = fu_engine_get_devices (priv->engine, error);
+	if (devs == NULL)
+		return FALSE;
+	state = fwupd_build_device_state_json (devs, error);
+	if (state == NULL)
+		return FALSE;
+	umask (0117);
```

Comment:
what for?

Reply:
`state.json` has the serial number, so it shouldn't be permissively written was my thought

Reply:
How could the session then read it? I think we want to support both, using perhaps `FWUPD_DEVICE_FLAG_TRUSTED` again.

Reply:
Hmm, that's true.  The problem with `FWUPD_DEVICE_FLAG_TRUSTED` is that `fwupdtool` must run as root (or a user with appropriate permissions) to be useful.  I guess I'll remove the umask and let this decision be made by the launcher of fwupdtool.